### PR TITLE
Create generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md

### PR DESCRIPTION
**Pull request by allipatev**
Add article "Generating TLS files yourself to avoid providing a fingerprint"

### CODE REVIEW

File: Environment-Management/generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+* [An Introduction to TLS](https://github.com/exasol/exasol-java-tutorial/blob/main/tls-tutorial/doc/tls_introduction.md)
+* [TLS With Exasol](https://github.com/exasol/exasol-java-tutorial/blob/main/tls-tutorial/doc/tls_with_exasol.md)
+
+Please acquaint yourself with the process of using and uploading TLS certificates for your database version, as well as with the description of the TLS encryption feature, using the links above.
+
+The tutorial applies to a particular use case:
+
+1. Only a connection string with IP addresses or DNS names could be used, fingerprint couldn't be added.
+2. Customer's IT Department or Security Department couldn't purchase a certificate from a public certification authority (CA) and also couldn't generate certificate in-house and sign it with company's customr root CA certificate.
```
Comment (allipatev) [inline Environment-Management/generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md:24]:
```suggestion
2. Customer's IT Department or Security Department couldn't purchase a certificate from a public certification authority (CA) and also couldn't generate certificate in-house and sign it with company's customer root CA certificate.
```

File: Environment-Management/generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+openssl req -x509 -new -nodes -key rootCA_key.pem -sha256 -days 3650 -out rootCA_cert.pem -subj '/CN=Custom_root_CA_for_Exasol_databases/O=test/C=DE'
+```
+
+This command creates a CA certificate as file `rootCA_cert.pem` with the following settings
+
+* Valid for 10 years (`-days 3650`).
+* The key is not password protected (`-nodes`)
+* Common name (`CN` in `-subj`) is "Custom root CA for Exasol databases"
+* Organization Name (`O` in `-subj`) is "test"
+* Country Name (`C` in `-subj`) is "test"
```
Comment (allipatev) [inline Environment-Management/generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md:70]:
```suggestion
* Country Name (`C` in `-subj`) is "DE"
```

File: Environment-Management/generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+```
+
+and uploading the resulting certificate chain file `cert_chain.pem` and key file `server_key.pem` via respective database interfaces.
+
+## Configuration of selected client applications
+
+In this section we'll show how to configure some client tools to work with the certificates we've just created.
+One needs to do so as the generated root CA certificate is not known to client applications.
+
+### DBeaver, DB Visualizer and other third party tools using Exasol JDBC driver
```
Comment (allipatev) [inline Environment-Management/generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md:121]:
```suggestion
### DBeaver, DbVisualizer and other third party tools using Exasol JDBC driver
```

File: Environment-Management/generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+Third party SQL clients leveraging Exasol JDBC driver typically come with their own JRE (Java Runtime Environment).
+By design Java applications use their own truststore and check if they trust database certificates by searching it in such a truststore.
+For example, as of 2024-11-25 for DBeaver (steps and paths for DB Visualizer are similar) JRE resides in `jre` subfolder of installation folder.
```
Comment (allipatev) [inline Environment-Management/generating-tls-files-yourself-to-avoid-providing-a-fingerprint.md:125]:
```suggestion
For example, as of 2024-11-25 for DBeaver (steps and paths for DbVisualizer are similar) JRE resides in `jre` subfolder of installation folder.
```

**Review by vshilin — APPROVED** [review]
great article

**Pull request merged**